File: Backend/barbershop/barbershop/Models/ResponseDTOs/ServiceDTO.cs
using barbershop.Models.Entitys;
using System;

namespace barbershop.Models.ResponseDTOs
{
    public class ServiceDTO
    {
        public int ServiceId { get; set; }

        public string ServiceName { get; set; } = null!;

        public int ServiceTypeId { get; set; }

        public string? Description { get; set; }

        public decimal Price { get; set; }

        public int DurationMin { get; set; }

        public bool IsActive { get; set; }

        public string? ServiceImage { get; set; }

        public bool? IsOutstanding { get; set; }
        public List<AppointmentServiceDTO> AppointmentServicesDTO { get; set; } = null!;
        public List<PaymentServicesDTO> PaymentServicesDTO { get; set; } = null!;

        public List<ServiceTypeDTO> ServiceTypeDTO { get; set; } = null!;
    }
}

File: Frontend/barbershop/src/types/ResponseDTOs/serviceDTO.ts
import type { AppointmentServiceDTO } from './appointmentServiceDTO';
import type { PaymentServiceDTO } from './paymentServiceDTO';
import type { ServiceTypeDTO } from './serviceTypeDTO';
export type ServiceDTO = {
    serviceId: number;
    serviceName: string;
    serviceTypeId: number;
    description: string;
    price: number;
    durationMin: number;
    isActive: boolean;
    serviceImage: string;
    isOutstanding: boolean;
    appointmentServicesDTO: AppointmentServiceDTO[];
    paymentServicesDTO: PaymentServiceDTO[];
    serviceTypeDTO: ServiceTypeDTO[];
};
